skorch/callbacks.py

```diff
@@ -365,3 +365,79 @@ def on_epoch_end(self, net, **kwargs):
         self._sink(tabulated.rsplit('\n', 1)[-1], verbose)
         if self.sink is print:
             sys.stdout.flush()
+
+
+class Checkpoint(Callback):
+    """Save the model during training if the given metric improved.
+
+    This callback works by default in conjunction with the validation
+    scoring callback since it creates a `valid_loss_best` value
+    in the history which the callback uses to determine if this
+    epoch is save-worthy.
+
+    You can also specify your own metric to monitor or supply a
+    callback that dynamically evaluates whether the model should
+    be saved in this epoch.
+
+    Example:
+
+    >>> net = MyNet(callbacks=[Checkpoint()])
+    >>> net.fit(X, y)
+
+    Example using custom monitor:
+
+    >>> monitor = lambda net: all(net.history[-1, (
+    ...     'train_loss_best', 'valid_loss_best')]
+    >>> net = MyNet(callbacks=[Checkpoint(monitor=monitor)])
+    >>> net.fit(X, y)
+
+    Parameters
+    ----------
+    target : file-like object, str
+      File path to the file or file-like object.
+      See NeuralNet.save_params for details what this value may be.
+
+      If the value is a string you can also use format specifiers
+      to, for example, indicate the current epoch. Accessible format
+      values are `net`, `last_epoch` and `last_batch`.
+      Example to include last epoch number in file name:
```

Comment:
Is there precedence for this? I would not have expected this behavior (which is not to say it isn't useful).

Reply:
Yes. See https://keras.io/callbacks/#modelcheckpoint